cg/meta/workflow/fluffy.py

```diff
@@ -0,0 +1,243 @@
+        return Path(self.root_dir, case_id, "output", "sacct", "submitted_jobs.yaml")
+
+    def get_priority(self, case_id: str) -> str:
+        """Returns priority for the case in clinical-db as text"""
+        case_object = self.status_db.family(case_id)
+        if case_object:
+            if case_object.high_priority:
+                return "high"
+            if case_object.low_priority:
+                return "low"
+        return "normal"
+
+    def get_sample_name_from_lims_id(self, lims_id: str) -> str:
+        LOG.debug("Setting sample name for %s", lims_id)
+        sample_obj = self.status_db.sample(lims_id)
+        return sample_obj.name
+
+    def link_fastq_files(self, case_id: str, dry_run: bool) -> None:
+        """
+        1. Get fastq from HK
+        2. Copy sample fastq to root_dir/case_id/fastq/sample_id (from samplesheet)
+        """
+        case_obj = self.status_db.family(case_id)
+        workdir_path = self.get_workdir_path(case_id=case_id)
+        if workdir_path.exists() and not dry_run:
+            LOG.info("Fastq directory exists, removing and re-linking files!")
+            shutil.rmtree(workdir_path, ignore_errors=True)
+        for familysample in case_obj.links:
+            sample_id = familysample.sample.internal_id
+            files = self.housekeeper_api.files(bundle=sample_id, tags=["fastq"])
+            sample_path: Path = self.get_fastq_path(case_id=case_id, sample_id=sample_id)
+            for file in files:
+                if not dry_run:
+                    Path.mkdir(sample_path, exist_ok=True, parents=True)
+                    Path(sample_path / Path(file.full_path).name).symlink_to(file.full_path)
+                LOG.info(f"Linking {file.full_path} to {sample_path / Path(file.full_path).name}")
+
+    def get_concentrations_from_lims(self, sample_id: str) -> str:
+        """Get sample concentration from LIMS"""
+        return self.lims_api.get_sample_attribute(lims_id=sample_id, key="concentration_sample")
+
+    def add_concentrations_to_samplesheet(
+        self, samplesheet_housekeeper_path: Path, samplesheet_workdir_path: Path
+    ) -> None:
+        csv_reader = csv.reader(open(samplesheet_housekeeper_path, "r"))
+        csv_writer = csv.writer(open(samplesheet_workdir_path, "w"))
+
+        sample_name_index = None
+        sample_id_index = None
+        for row in csv_reader:
+            if "SampleID" in row and not sample_id_index:
+                sample_name_index = row.index("SampleName")
+                sample_id_index = row.index("SampleID")
+                row.append("Library_nM")
+                csv_writer.writerow(row)
+                continue
+            if sample_id_index:
+                sample_id = row[sample_id_index]
+                row[sample_name_index] = self.get_sample_name_from_lims_id(lims_id=sample_id)
+                row.append(str(self.get_concentrations_from_lims(sample_id=sample_id)))
+                csv_writer.writerow(row)
```

Comment:
```suggestion
            if sample_id_index:
                sample_id = row[sample_id_index]
                row[sample_name_index] = self.get_sample_name_from_lims_id(lims_id=sample_id)
                row.append(str(self.get_concentrations_from_lims(sample_id=sample_id)))
                csv_writer.writerow(row)
                continue
            if "SampleID" in row:
                sample_name_index = row.index("SampleName")
                sample_id_index = row.index("SampleID")
                row.append("Library_nM")
                csv_writer.writerow(row)
```

Reply:
This has to work in this specific way because we have to keep skipping rows in csv until we find header with SampleID, and then for the following rows we can start using values in that column to fetch lims data